### PR TITLE
UI improvement

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: identity-composer-component.html [hunk 1]
```diff
@@ -1,31 +1,137 @@
...
+
 <div ng-if="$ctrl.loading" style="width: 100vw; font-size: x-small;">
   <i class="fa fa-refresh fa-spin text-muted"></i> Building profiles...
 </div>
 <div ng-if="!$ctrl.loading" ng-style="$ctrl.profileContainerStyle">
   <div ng-if="!!$ctrl.profiles.length" class="br-identity-profile-list"
     ng-repeat="profile in $ctrl.profiles">
-    <h4 ng-mouseenter="$hover=true" ng-mouseleave="$hover=false"
-      ng-click="$ctrl.select(profile, $index)">
+<!--     <h4 ng-mouseenter="$hover=true" ng-mouseleave="$hover=false">
```
Comment (mattcollier) [inline identity-composer-component.html]: Remove commented code if not needed.

File: identity-composer-component.html [hunk 2]
```diff
@@ -1,31 +1,137 @@
...
+    <div class="text-container">
+      <div class="text-element"
+        style="font-size: calc({{$ctrl.credentialWidth.select}} * 0.07)">
         Select Profile {{$index + 1}}
         <i ng-if="$ctrl.sending && $ctrl.selectedIndex === $index"
-          class="fa fa-refresh fa-spin"></i>
-      </span>
-    </h4>
-    <div class="br-identity-profile-container"
+            class="fa fa-refresh fa-spin"></i>
```
Comment (mattcollier) [inline identity-composer-component.html]: Looks like your editor needs to be configured to remove trailing spaces.

File: identity-composer-component.js [hunk 3]
```diff
@@ -144,6 +144,13 @@ function Ctrl($q, brCredentialLibraryService, brMediaQueryService) {
       });
   };
 
+  self.detail = function(credential, showCredential, $event) {
+    $event.stopPropagation();
+    self.credential = credential;
+    self.showCredentialDetails = showCredential;
+
```
Comment (mattcollier) [inline identity-composer-component.js]: Remove extra line.

File: identity-composer-component.html [hunk 4]
```diff
@@ -1,31 +1,137 @@
+<style>
```
Comment (mattcollier) [inline identity-composer-component.html]: I guess this would be moved into a `less` file for this module once everyone signs off on the design.

**Review by mattcollier — CHANGES_REQUESTED** [review]

File: identity-composer-component.html (hunk 4 above)
Comment (dlongley) [reply identity-composer-component.html]: All of these special classes should be prefixed as well with `br-` -- CSS classes are global.

File: identity-composer-component.html [hunk 5]
```diff
@@ -1,31 +1,137 @@
...
         <!-- TODO: ng-class="alert alert-warning"/"alert-danger for superfluous
           and sensitive properties -->
         <!-- TODO: change simple to complex displayer -->
-        <br-credential br-credential="credential" br-show-actions="false"
-          br-options="{displayer: {style: {width: $ctrl.credentialWidth.select}}}">
-        </br-credential>
+        <div class="card-wrapper" style="width: {{$ctrl.credentialWidth.select}}">
+          <div class="icon-wrapper"
+              style="width: calc({{$ctrl.credentialWidth.select}} * 0.0975);
+                      height: calc({{$ctrl.credentialWidth.select}} * 0.0975)">
```
Comment (dlongley) [inline identity-composer-component.html]: Fix indentation. Also, CSS `calc` wouldn't have to be relied upon if these calculations were performed in the controller and stored in some object for the info overlay. The factor `0.0975` could also be in the controller as a constant. Also, why is that multiplier sometimes `0.0975` and `0.10` at other points? Can something be done to make that seem less magical and easier to understand why it is what it is?

File: identity-composer-component.html [hunk 6]
```diff
@@ -1,31 +1,137 @@
...
-      ng-click="$ctrl.select(profile, $index)">
+<!--     <h4 ng-mouseenter="$hover=true" ng-mouseleave="$hover=false">
       <span class="label label-primary" style="cursor: pointer;">
+        
+      </span>
+    </h4> -->
+ 
+    <div class="text-container">
+      <div class="text-element"
+        style="font-size: calc({{$ctrl.credentialWidth.select}} * 0.07)">
```
Comment (davidlehn) [inline identity-composer-component.html]: Could also use angular to do the math vs using `calc()`.
Should `ng-style` be used here?

File: identity-composer-component.js [hunk 7]
```diff
@@ -144,6 +144,13 @@ function Ctrl($q, brCredentialLibraryService, brMediaQueryService) {
       });
   };
 
+  self.detail = function(credential, showCredential, $event) {
```
Comment (dlongley) [inline identity-composer-component.js]: Naming convention for functions called in response to an event/action is: `onFoo`. So this should be something like: `onInfoClicked` or `onShowDetails` or similar.

**Comment by dlongley** [comment]
ping @gannan08: changes requested here

File: main.less [hunk 8]
```diff
@@ -20,7 +20,7 @@
   border-radius: 0.25em;
   display: flex;
   flex-flow: row wrap;
-  padding: 10px 5px;
+  padding: 3.5% 5px;
```
Comment (dlongley) [inline main.less]: This seems problematic, as in, lots of padding on large screens. Why not fixed padding and allow parent layouts that are using this class to deal with padding complexity?

File: main.less [hunk 9]
```diff
@@ -59,3 +58,65 @@
...
+    left: -35%;
+    font-size: 1.6em;
+    font-weight: bold;
+  }
+}
+@media (max-width: 767px) {
+  .br-text-element {
+    left: -50%;
+  }
+}
```
Comment (dlongley) [inline main.less]: Combine media queries into `br-text-element` (less syntax):

```css
.br-text-element {
  ...

  @media (max-width: 1440px) {
  }
  ...
}

File: main.less [hunk 10]
```diff
@@ -59,3 +58,65 @@
...
+    margin-left: 10%;
+  }
+}
+@media (max-width: 767px) {
+  .br-text-container {
+    position: absolute;
+    margin-left: 50%;
+    margin-top: 1%;
+  }
+}
```
Comment (dlongley) [inline main.less:91]: Combine (group) media queries like suggested above.

File: main.less [hunk 11]
```diff
@@ -59,3 +58,65 @@
...
+  background-color: rgba(255, 255, 255, 0.6);
+  position: absolute;
+  width: 20px;
+  height: 20px;
+  border-radius: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+@media (max-width: 767px) {
```
Comment (dlongley) [inline main.less]: Move up to be in the rest of the `br-identity-container` definition.

File: identity-composer-component.js [hunk 12]
```diff
@@ -144,6 +144,19 @@ function Ctrl($q, brCredentialLibraryService, brMediaQueryService) {
       });
   };
 
+  self.scaleWidth = function(scale) {
+    // remove the vw unit
+    var width = self.credentialWidth.select.slice(0,-2);
+    var scaledWidth = (scale * width).toFixed(2);
+    return scaledWidth.toString() + 'vw';
+  };
```
Comment (dlongley) [inline identity-composer-component.js:153]: This seems kinda hacky to me ... but we can move forward with it and address later. Please mark with a "TODO: could be improved, seems hacky" so we can note it for later :)

**Review by dlongley — CHANGES_REQUESTED** [review]

**Review by dlongley — APPROVED** [review]

**Review by mattcollier — APPROVED** [review]

**Pull request merged**